api/external_resources/models.py

```diff
@@ -0,0 +1,111 @@
+
+            for feature_state in feature_states:
+                feature_env_data = {}
+                if feature_state.feature_state_value.string_value is not None:
+                    feature_env_data["string_value"] = (
+                        feature_state.feature_state_value.string_value
+                    )
+                if feature_state.feature_state_value.boolean_value is not None:
+                    feature_env_data["boolean_value"] = (
+                        feature_state.feature_state_value.boolean_value
+                    )
+                if feature_state.feature_state_value.integer_value is not None:
+                    feature_env_data["integer_value"] = (
+                        feature_state.feature_state_value.integer_value
+                    )
+
+                feature_env_data["environment_name"] = feature_state.environment.name
+                feature_env_data["feature_value"] = feature_state.enabled
+                if (
+                    hasattr(feature_state, "feature_segment")
+                    and feature_state.feature_segment is not None
+                ):
+                    feature_env_data["segment_name"] = (
+                        feature_state.feature_segment.segment.name
+                    )
+                feature_data["feature_states"].append(feature_env_data)
+
+            call_github_app_webhook_for_feature_state.delay(
+                args=(
+                    feature_data,
+                    WebhookEventType.FEATURE_EXTERNAL_RESOURCE_ADDED.value,
+                ),
+            )
+        else:
+            organisation_id = self.feature.project.organisation.id
+            print(
+                f"No GitHub integration exists for organisation {organisation_id}. Not calling webhooks."
+            )
+
+    @hook(BEFORE_DELETE)
+    def create_feature_external_resource_removed_comment(self):
+        if hasattr(self.feature.project.organisation, "github_config"):
+            github_configuration = self.feature.project.organisation.github_config
+            feature_data = {
+                "id": self.feature_id,
+                "name": self.feature.name,
+                "url": self.url,
+            }
+            feature_data["installation_id"] = github_configuration.installation_id
+            feature_data["organisation_id"] = github_configuration.organisation.id
+
+            call_github_app_webhook_for_feature_state.delay(
+                args=(
+                    feature_data,
+                    WebhookEventType.FEATURE_EXTERNAL_RESOURCE_REMOVED.value,
+                ),
+            )
+        else:
+            organisation_id = self.feature.project.organisation.id
+            print(
+                f"No GitHub integration exists for organisation {organisation_id}. Not calling webhooks."
+            )
+
+    class Meta:
+        ordering = ("id",)
+        unique_together = ("url",)
```

Comment:
I will check the unique constraint.

On the other hand, these constraints only ensure that the same GitHub Issue or PR URL is linked to the same feature, it does not restrict features from being linked to different issues or PRs.

Nor does it prevent multiple features from being linked to the same Issue or PR.

Reply:
You were right, this should be a compound constraint on the fields `feature` and `url` since we want to allow any URL to be linked to several Features. Corrected.